lib/elixir/lib/float.ex

```diff
@@ -365,6 +365,7 @@ defmodule Float do
         case rounding do
           :ceil when sign === 0 -> 1 / power_of_10(precision)
           :floor when sign === 1 -> -1 / power_of_10(precision)
+          :ceil when sign === 1 -> -0.0
```

Comment:
Fixing `Float.ceil/2` to match `Float.ceil/1` and `:math.ceil/1`

Reply:
Actually I'm not sure, we might want to rethink this whole rounding spec.

Here is the current behavior:

```elixir
Erlang/OTP 26 [erts-14.2.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Interactive Elixir (1.16.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> for fun <- [:round, :ceil], num <- [-0.0, -0.01], prec <- 0..1, do: IO.puts "#{fun}(#{num}), #{prec}) = #{apply(Float, fun, [num, prec])}"
round(-0.0), 0) = 0.0
round(-0.0), 1) = -0.0
round(-0.01), 0) = 0.0
round(-0.01), 1) = 0.0
ceil(-0.0), 0) = -0.0
ceil(-0.0), 1) = -0.0
ceil(-0.01), 0) = -0.0
ceil(-0.01), 1) = 0.0
```

It feels like we'd want to have to following invariants:
- `ceil(x) >= round(x)`
- `fun(-0.01) <= fun(-0.0)`
- `fun(-0.01, 0) === fun(-0.01, 1)`
but none of these are currently respected.

How about the following?

```elixir
round(-0.0), 0) = 0.0
round(-0.0), 1) = 0.0
round(-0.01), 0) = 0.0
round(-0.01), 1) = 0.0
ceil(-0.0), 0) = 0.0
ceil(-0.0), 1) = 0.0
ceil(-0.01), 0) = 0.0
ceil(-0.01), 1) = 0.0
```

Reply:
Looks excellent to me, awesome summary and proposal.

Reply:
OK should be good! https://github.com/elixir-lang/elixir/pull/13351/commits/5edfdd6d19b1a6c1a7255ec5239502e5d52d70f6